spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -394,14 +393,15 @@ private boolean sendWithObservation(Message<?> message, long timeout) {
 				messageToSendInternal =
 						new ErrorMessage(errorMessage.getPayload(),
 								messageToSend.getHeaders(),
-								errorMessage.getOriginalMessage());
+								Objects.requireNonNull(errorMessage.getOriginalMessage()));
 			}
 			return sendInternal(messageToSendInternal, timeout);
 		});
 		return Boolean.TRUE.equals(observe);
 	}
 
 	private boolean sendWithMetrics(Message<?> message, long timeout) {
+		Assert.state(this.metricsCaptor != null, "The metricsCaptor must not be null");
```

Comment:
Better to use `@SuppressWarnings("NullAway") // Dataflow analysis limitation` on this method instead.
The point is that we don't go to this method at all if `this.metricsCaptor` is `null`.
On the other side this is a critical path of the whole messaging, and having an extra call in this method for `Assert` might cause a performance impact.